python/pyarrow/_fs.pyx

```diff
@@ -335,6 +432,22 @@ cdef class FileSystem(_Weakrefable):
         tuple of (FileSystem, str path)
             With (filesystem, path) tuple where path is the abstract path
             inside the FileSystem instance.
+
+        Examples
+        --------
+        Create a new FileSystem subclass from a URI:
+
+        >>> uri = 'file:///{}/pyarrow-fs-example.dat'.format(local_path)
```

Comment:
Actually, we don't need to format it with a "local_path" variable, as this URI can be any random path (the method doesn't verify that the path exists, only parses it):

```
In [34]: fs.FileSystem.from_uri("file:///home/user/example.dat")
Out[34]: (<pyarrow._fs.LocalFileSystem at 0x7f269d9cc530>, '/home/user/example.dat')
```

Reply:
(although of course this pattern is also used in the examples below, so maybe just fine to do here as well ..)